### PR TITLE
feat: Add "rentedDay" and "periodChosen" columns

### LINKED ISSUE #163: [BUG] The /GET method for rentals is not returning the `periods` array in order
When getting the listings, some listings are returning the periods without the proper `period` orders.

### LINKED ISSUE #164: Make the transactions server store the period and the days that the user selected
As of today, users create rentals specifying one of the multiple periods that define the cost per day of the rental and the  restrictions that the rental will have in terms of days (min and max days). With that, the user also specifies an amount of days that fit the restrictions of the selected period, min days <= selected days <= max days.
The signature server is not currently storing the amount of days and the period that the user selected, preventing us to show it in the UI.
To fix this:
- Change the database to store the selected days and the selected period.
- Change the refresh endpoint to retrieve the selected period and days from the rentals graph and save it into the rental.
- Change the update job to update the rentals tables with the data.

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Closes #163
Closes #164

### CODE REVIEW

File: src/migrations/1667818838202_add-rentedDays-and-periodChosen-columns.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MigrationBuilder, ColumnDefinitions } from "node-pg-migrate"
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+const tableName = "rentals"
+const rentedDaysColumn = "rentedDays"
```
Comment (LautaroPetaccio) [inline src/migrations/1667818838202_add-rentedDays-and-periodChosen-columns.ts:7]: We're using snake case for the columns and tables, would you mind changing it?
```suggestion
const rentedDaysColumn = "rented_days"
```

File: src/ports/rentals/component.ts [hunk 2]
```diff
@@ -452,9 +452,12 @@ export async function createRentalsComponent(
...
       status: RentalStatus
       lessor: string
+      period_id: number
+      max_days: number
+      min_days: number
     }>(
-      SQL`SELECT rentals.id, rentals.contract_address, rentals.token_id, rentals.updated_at, rentals.signature, rentals.nonces, rentals.status, metadata.id as metadata_id, metadata.updated_at as metadata_updated_at, rentals_listings.lessor as lessor
-      FROM rentals, metadata, rentals_listings
+      SQL`SELECT rentals.id, rentals.contract_address, rentals.token_id, rentals.updated_at, rentals.signature, rentals.nonces, rentals.status, metadata.id as metadata_id, metadata.updated_at as metadata_updated_at, rentals_listings.lessor as lessor, p.id period_id, p.max_days, p.min_days 
+      FROM rentals JOIN periods p on p.rental_id = rentals.id, metadata, rentals_listings
```
Comment (LautaroPetaccio) [inline src/ports/rentals/component.ts:460]: Was there a requirement to use the `JOIN` keyword? The keyword seems to have divided the clauses that were being used to join the tables, `metadata.id = rentals.metadata_id` was left on the WHERE clause and is part of the JOIN.

File: src/ports/rentals/component.ts [hunk 3]
```diff
@@ -683,23 +688,34 @@ export async function createRentalsComponent(
...
+                WHERE rentals.id = rentals_listings.id AND rentals.signature = ${rental.signature}
+              `
             )
             logger.debug(
-              `[Rentals update][Exists:${Boolean(dbRental)}][contractAddress:${rental.contractAddress}][tokenId:${
+              `[Rentals update][Exists:${Boolean(dbRentals[0])}][contractAddress:${rental.contractAddress}][tokenId:${
                 rental.tokenId
               }]`
             )
+            const dbRental = dbRentals.find(({ max_days }) => max_days === Number(rental.rentalDays)) || dbRentals[0]
```
Comment (LautaroPetaccio) [inline src/ports/rentals/component.ts:709]: Is this used to find a `dbRental` from all the possible dbRentals? There can't be more than one dbRental with the same signature, so this shouldn't happen.

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/migrations/1667818838202_add-rentedDays-and-periodChosen-columns.ts (hunk 1 above)
Comment (juanmahidalgo) [reply src/migrations/1667818838202_add-rentedDays-and-periodChosen-columns.ts]: Updated!

File: src/ports/rentals/component.ts (hunk 2 above)
Comment (juanmahidalgo) [reply src/ports/rentals/component.ts]: Not sure I'm following the requested change here, I think the query was working as expected, the `WHERE` was not changed, I just added an _explicit_ `JOIN`. I changed to an _implicit_ one to keep the same approach as the other ones.

File: src/ports/rentals/component.ts (hunk 3 above)
Comment (juanmahidalgo) [reply src/ports/rentals/component.ts:709]: The query now that is joining by period, will return several rows and so we need to know which period corresponds with the `rentalDays`

File: src/ports/rentals/component.ts [hunk 4]
```diff
@@ -683,23 +688,36 @@ export async function createRentalsComponent(
...
+            // we will need to re-evaluate this logic
+            const dbRental = dbRentals.find(({ max_days }) => max_days === Number(rental.rentalDays)) || dbRentals[0]
             // If there's a rental in the database updated it, else, create it
             if (dbRental) {
               await Promise.all([
                 client.query(
                   SQL`UPDATE rentals SET updated_at = ${new Date(
                     fromSecondsToMilliseconds(Number(rental.updatedAt))
-                  )}, started_at = ${new Date(fromSecondsToMilliseconds(Number(rental.startedAt)))}, status = ${
+                  )}, rented_days = ${rental.rentalDays}, period_chosen = ${
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/ports/rentals/component.ts:718]: Not needed.

File: src/ports/rentals/component.ts [hunk 5]
```diff
@@ -502,6 +505,8 @@ export async function createRentalsComponent(
         database.query(
           SQL`UPDATE rentals SET updated_at = ${new Date(indexerRentalLastUpdate)}, status = ${
             indexerRentals[0].ownerHasClaimedAsset ? RentalStatus.CLAIMED : RentalStatus.EXECUTED
+          }, rented_days = ${indexerRentals[0].rentalDays}, period_chosen = ${
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/ports/rentals/component.ts:508]: Not needed.

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**